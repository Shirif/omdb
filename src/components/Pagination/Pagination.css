.pagination {
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  color: #d3d3d3;
  padding: 25px 0 10px;
  margin: 0;
  border: none;
  font-size: 1.2rem;
}

.pagination li {
  color: #d3d3d3;
  background-color: #000000;
  box-shadow: 0 0 20px 2px #3d3d3d;
  float: left;
  padding: 5px 12px;
  text-decoration: none;
  transition: background-color 0.4s;
  margin: 2px;
  cursor: pointer;
}

.pagination li.active {
  background-color: #d3d3d3;
  color: #000000;
}

.pagination li:hover:not(.active) {
  background-color: #3fff05;
  color: #000000;
}
ul {
  list-style-type: none;
}
