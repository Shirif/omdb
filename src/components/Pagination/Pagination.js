import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { setPageSearchData, fetchMovies } from "../../store/actionCreators/action";

import "./Pagination.css";

const Pagination = ({ totalResults }) => {
  const dispatch = useDispatch();
  const searchData = useSelector((state) => state.movies.searchData);
  const pageSize = 10;

  const pagesCount = Math.ceil(+totalResults / pageSize);
  const pages = [];
  for (let i = 1; i <= pagesCount; i++) {
    pages.push(i);
  }

  const setCurrentPage = (e) => {
    const page = e.target.innerText;
    dispatch(setPageSearchData(page));
    dispatch(fetchMovies({ title: searchData.title, year: searchData.year, page }));
  };

  const listElements = pages.map((p) => {
    return (
      <li searchData={searchData} className={+searchData.page === p ? "active" : ""} key={p} onClick={setCurrentPage}>
        {p}
      </li>
    );
  });

  return <ul className="pagination">{listElements}</ul>;
};

export default Pagination;
