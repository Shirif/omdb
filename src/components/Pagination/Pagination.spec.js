import { shallow } from "enzyme";
import React from "react";
import * as reactRedux from "react-redux";
import Pagination from "./Pagination";

const setUp = () => shallow(<Pagination />);

describe("should render Pagination component", () => {
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  let component;
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
   });

  it("should contain .pagination", () => {
    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);
    //expect(dummyDispatch).not.toHaveBeenCalled();
    useSelectorMock.mockReturnValue({});
    component = setUp();
    const wrapper = component.find(".pagination");
    expect(wrapper.length).toBe(1);
  });

  it("should render pagination button", () => {
    const props = 50;
    component = shallow(<Pagination totalResults={props} />);
    const wrapper = component.find("li");
    expect(wrapper.length).toBe(props / 10);
  });


  it("calls click button", () => {
    useSelectorMock.mockReturnValue({page: "1"});
    const props = 50;
     
    component = shallow(<Pagination totalResults={props}/>);
    const wrapper = component.find("li").first().simulate("click", {
      target: {
        innerText: 1,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
