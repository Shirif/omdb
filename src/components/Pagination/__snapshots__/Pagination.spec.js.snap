// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination component calls click button 1`] = `
<li
  className="active"
  key="1"
  onClick={[Function]}
  searchData={
    Object {
      "page": "1",
    }
  }
>
  1
</li>
`;
